src/app/PluginDependencyFlags.h

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/config.h>
+
+#if defined(MATTER_DM_PLUGIN_ON_OFF)
+#undef MATTER_DM_PLUGIN_ON_OFF
+#define MATTER_DM_PLUGIN_ON_OFF 1
+#else
+#define MATTER_DM_PLUGIN_ON_OFF 0
```

Comment:
Instead of working around what the codegen spits out, could we not fix coden (to output new "always defined" macros that have this information) and then switch to the new codegen bits?